power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -59,20 +175,29 @@ template <dataset_type_tag dataset_type_> class Dataset {
     template <class StructType>
     using DataStruct = std::conditional_t<is_data_mutable_v<dataset_type>, StructType, StructType const>;
 
+    struct AttributeBuffers {
+        std::vector<Data*> data;
+        std::vector<std::reference_wrapper<MetaAttribute const>> meta_attributes;
+    };
+
+    // for columnar buffers, Data* data is empty and attributes is filled
+    // for uniform buffers, indptr is empty
     struct Buffer {
         Data* data;
-        // for uniform buffer, indptr is empty
+        AttributeBuffers attributes;
         std::span<Indptr> indptr;
+
+        Buffer(Data* data = nullptr, std::span<Indptr> indptr = {}) : data(data), indptr(indptr) {}
     };
 
+    template <class StructType>
+    using RangeObject = std::conditional_t<is_data_mutable_v<dataset_type>, mutable_range_object<StructType>,
+                                           const_range_object<StructType>>;
+
     static constexpr Idx invalid_index{-1};
 
     Dataset(bool is_batch, Idx batch_size, std::string_view dataset_name, MetaData const& meta_data)
-        : meta_data_{&meta_data},
-          dataset_info_{.is_batch = is_batch,
-                        .batch_size = batch_size,
-                        .dataset = &meta_data.get_dataset(dataset_name),
-                        .component_info = {}} {
+        : meta_data_{&meta_data}, dataset_info_{is_batch, batch_size, &meta_data.get_dataset(dataset_name)} {
```

Comment:
because you have now a constructor, we cannot use the nice named attributes anymore.

Reply:
Reverted to old after removing constructors.